cameraview/src/main/java/com/otaliastudios/cameraview/VideoResult.java

```diff
@@ -133,6 +137,16 @@ public File getFile() {
         return file;
     }
 
+    /**
+     * Returns the file descriptor where the video was saved.
+     *
+     * @return the File Descriptor of this video
+     */
+    @NonNull
+    public FileDescriptor getFileDescriptor() {
+        return fileDescriptor;
+    }
+
```

Comment:
I don't like the fact that both this and getFile are marked as `@NonNull`, but both can be null...
Making both `@Nullable` would be ideal but it would break people's code on Kotlin.

I think that we should leave `@NonNull` but throw in both getFileDescriptor and getFile.

```
    @NonNull
    public FileDescriptor getFileDescriptor() {
        if (fileDescriptor == null) {
            throw new RuntimeException("FileDescriptor is only available when takeVideo(FileDescriptor) is used.")
        }
        return fileDescriptor;
    }
```

Reply:
Oh right, I missed that!